src/top.c

```diff
@@ -249,7 +249,7 @@ static void redraw(void) {
 
 	if(changed) {
 		n = 0;
-		sorted = xrealloc(sorted, node_list.count * sizeof(*sorted));
+		sorted = xrealloc(sorted, node_list.count * sizeof(sorted));
```

Comment:
The original `sizeof(*sorted)` is correct, sorted itself is just the pointer to the array, we want the size of an array element here.

Reply:
It's a pointer to a pointer, though? I checked it with `_Static_assert` before pushing just to make sure. Both of these pass:

```c
_Static_assert(sizeof(*sorted) == sizeof(void*), "foo");
_Static_assert(sizeof(nodestats_t) != sizeof(void*), "foo");
```

Reply:
Yes, but if the type of `sorted` is ever changed, then `sizeof(sorted)` might become really wrong. Was clang-tidy complaining about this?

Reply:
I don't like adding comments to appease linters, it quickly becomes a mess otherwise. You already disabled several checks, I would disable this one as well if it generates false positives.

Reply:
Ah, ok. Makes sense. Although those are disabled temporarily (hopefully) as especially problematic. Disabling one or two rules allows us to enable 10-20 others by globbing the whole group. I expect to fix most of them later.